plugins/legacy-plugin-chart-calendar/src/Calendar.js

```diff
@@ -117,7 +118,19 @@ function Calendar(element, props) {
       calContainer.text(`Metric: ${verboseMap[metric] || metric}`);
     }
     const timestamps = metricsData[metric];
-    const extents = d3Extent(Object.keys(timestamps), key => timestamps[key]);
+
+    const newData = Object.keys(timestamps).map(key => ({
+      timestamp: key,
+      value: timestamps[key],
+    }));
+
+    const objPeriodDay = groupByTimePeriod(newData, 'timestamp', subdomainGranularity);
+    const interval = Object.keys(objPeriodDay).map(key =>
+      objPeriodDay[key].reduce((a, b) => a + b.value, 0),
+    );
```

Comment:
This logic assumes that the metrics are additive - if we have a non-additive metric like `AVG`, this won't work correctly. My recommendation would be to apply the subdomain granularity to an actual time grain in the backend, thus forcing whichever subdomain has been chosen as the timegrain in the query. This would ensure that the metric is always aggregated correctly. Since this plugin resides in `viz.py`, it should be fairly simple to modify (I can help out if needed).